src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -12,37 +12,66 @@
 namespace Symfony\Component\Serializer\Exception;
 
 /**
- * @deprecated since Symfony 6.3, use {@see MissingConstructorArgumentException} instead
- *
  * @author Maxime VEBER <maxime.veber@nekland.fr>
  */
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
-     * @var string[]
+     * @var string|null
+     */
+    private $class;
+
+    /**
+     * @var MissingConstructorArgumentException[]
      */
-    private $missingArguments;
+    private $missingConstructorArgumentExceptions = [];
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    /**
+     * @param ?class-string $class
+     * @param MissingConstructorArgumentException[] $missingArguments
+     */
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, string $class = null, array $missingArguments = [])
```

Comment:
This is still a BC break, any new argument needs to be added to the end so that existing code wouldn't break, eg:

```php
new MissingConstructorArgumentsException('', 0, null, []); // this would now break
```

Also, I'd suggest removing the `MissingConstructorArgumentException` as it's no longer needed.